src/wallet/rpc/util.cpp

```diff
@@ -29,6 +29,27 @@ bool GetAvoidReuseFlag(const CWallet& wallet, const UniValue& param) {
     return avoid_reuse;
 }
 
+std::string EnsureUniqueWalletName(const JSONRPCRequest& request, const std::string* wallet_name)
```

Comment:
I wrote some unit tests for this function, feel free to add to this PR if you think it's helpful: https://github.com/stickies-v/bitcoin/commit/60f9dc8af66bbd4eda81049a10262527f83356a1

Reply:
Thanks! Makes sense, I'll add them.